### ISSUE TITLE
Can't launch py script with hass.io installation

### ISSUE THREAD

**Issue by Wopalecki** [OmenWild/home-assistant-z-wave-graph#6]
Hi Omen,

Thank you for your work. I tried to extend my hass.io installation with your z-wave-graph script. I followed the instructions in the README.md file.
Unfortunaly, it doesn't work, I received the following message in the Home Assistant logs (dev-info): 

Error running command: `/usr/bin/python3 /usr/bin/z-wave-graph.py --debug`, return code: 2
NoneType: None

Can you help me? Thanks in advance

**Comment by OmenWild** [comment]
Fortunately you are not the only one to run into this issue. See [here](https://community.home-assistant.io/t/graph-your-z-wave-mesh-python-auto-update/40549/70?u=omenwild) for a discussion about it on the HA forums. Unfortunately, we have not yet been able to solve it, there is something about that environment that makes the shell_command fail.

**Comment by OmenWild** [comment]
I released an update yesterday that verifies the login through the API worked correctly. Give it a try and let me know if it shows anything.

**Comment by Wopalecki** [comment]
Unfortunately, got the same error message.
And the following logs (debug mode) didn't give any information on the cause:

```
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.components.automation] Executing Generate Z-Wave graph
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.core] Bus:Handling <Event logbook_entry[L]: name=Generate Z-Wave graph, message=has been triggered, domain=automation, entity_id=automation.generate_zwave_graph>
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.helpers.script] Script Generate Z-Wave graph: Running script
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.helpers.script] Script Generate Z-Wave graph: Executing step call service
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=shell_command, service=z_wave_graph, service_data=, service_call_id=1965183152-4>
2018-02-02 10:35:00 ERROR (MainThread) [homeassistant.components.shell_command] Error running command: `/usr/bin/python3 /usr/bin/z-wave-graph.py --debug`, return code: 2
NoneType: None
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.core] Bus:Handling <Event service_executed[L]: service_call_id=1965183152-4>
2018-02-02 10:35:00 INFO (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=automation.generate_zwave_graph, old_state=<state automation.generate_zwave_graph=on; last_triggered=2018-02-02T09:30:00.517748+00:00, friendly_name=Generate Z-Wave graph @ 2018-02-02T10:32:10.394792+01:00>, new_state=<state automation.generate_zwave_graph=on; last_triggered=2018-02-02T10:35:00.797290+01:00, friendly_name=Generate Z-Wave graph @ 2018-02-02T10:32:10.394792+01:00>>
2018-02-02 10:35:00 DEBUG (MainThread) [homeassistant.components.websocket_api] WS 1925036208: Sending {'id': 2, 'type': 'event', 'event': {'event_type': 'state_changed', 'data': {'entity_id': 'automation.generate_zwave_graph', 'old_state': <state automation.generate_zwave_graph=on; last_triggered=2018-02-02T09:30:00.517748+00:00, friendly_name=Generate Z-Wave graph @ 2018-02-02T10:32:10.394792+01:00>, 'new_state': <state automation.generate_zwave_graph=on; last_triggered=2018-02-02T10:35:00.797290+01:00, friendly_name=Generate Z-Wave graph @ 2018-02-02T10:32:10.394792+01:00>}, 'origin': 'LOCAL', 'time_fired': datetime.datetime(2018, 2, 2, 9, 35, 0, 797895, tzinfo=<UTC>)}}
```

**Comment by OmenWild** [comment]
Someone in the [HA forum](https://community.home-assistant.io/t/graph-your-z-wave-mesh-python-auto-update/40549/87?u=omenwild) solved it. I pushed a new version with a code tweak. Make sure you check the `Docker Image` section of the README for an additional change you need in the shell_command section.

**Comment by wijnandtop** [comment]
I still had some troubles running this script inside hass.io even after the tweaks

First I did was fetching the output of the python script:

shell_commands/z-wave-graph.yaml
`z_wave_graph:  /usr/bin/python3 /config/python_scripts/z-wave-graph.py &> /config/zwave-graph.log`

Which resulted in an log entry which explained why it didn't work; "please pip install networkx"

As a quick fix I changed the zwave_graph command to (and changed it back after a single run):

`z_wave_graph:  pip3 install networkx &> /config/zwave-graph.log`

This of course needs a more sustainable fix, it now breaks after each hass.io update (which doesn't include networkx module in it docker image). Maybe there are some hass.io upgrade events which can be a trigger to execute the pip install...

**Comment by OmenWild** [comment]
HA has a mechanism where it auto-installs the modules it needs. I'll see if I can tap into it.

**Comment by OmenWild** [comment]
Someone got tired of waiting for me and added the code (#9), so this should be resolved.

Closes: #6
